Source/SA/DPD/DPD.cs

```diff
@@ -0,0 +1,207 @@
+                    SignalType = RFmxSpecAnMXDpdSignalType.Modulated,
+                    DutAverageInputPower = -20,
+                    PreDpdCFREnabled = RFmxSpecAnMXDpdPreDpdCfrEnabled.False,
+                    ApplyDpdCFREnabled = RFmxSpecAnMXDpdApplyDpdCfrEnabled.False
+                };
+            }
+
+        }
+        public struct LookupTableConfiguration
+        {
+            public RFmxSpecAnMXDpdLookupTableThresholdType ThresholdType;
+            public RFmxSpecAnMXDpdLookupTableThresholdEnabled ThresholdEnabled;
+            public double ThresholdLevel;
+            public double LookupTableStepSize;
+            public RFmxSpecAnMXDpdApplyDpdLookupTableCorrectionType LookupTableCorrectionType;
+            public RFmxSpecAnMXDpdLookupTableType LookupTableType;
+
+            public static LookupTableConfiguration GetDefault()
+            {
+                return new LookupTableConfiguration
+                {
+                    LookupTableType = RFmxSpecAnMXDpdLookupTableType.Linear,
+                    ThresholdType = RFmxSpecAnMXDpdLookupTableThresholdType.Relative,
+                    ThresholdEnabled = RFmxSpecAnMXDpdLookupTableThresholdEnabled.True,
+                    ThresholdLevel = -20,
+                    LookupTableStepSize = 0.1,
+                    LookupTableCorrectionType = RFmxSpecAnMXDpdApplyDpdLookupTableCorrectionType.MagnitudeAndPhase
+
+                };
+            }
+        }
+
+        public struct MemoryPolynomialConfiguration
+        {
+            public RFmxSpecAnMXDpdApplyDpdMemoryModelCorrectionType MemoryModelCorrectionType;
+            public RFmxSpecAnMXDpdIterativeDpdEnabled IterativeDpdEnabled;
+            public int NumberOfIteration;
+            public int MemoryPolynomialOrder, MemoryPolynomialDepth, MemoryPolynomialLeadOrder, MemoryPolynomialLagOrder,
+            MemoryPolynomialLeadMemoryDepth, MemoryPolynomialLagMemoryDepth, MemoryPolynomialMaxLead, MemoryPolynomialMaxLag;
+            public static MemoryPolynomialConfiguration GetDefault()
+            {
+                return new MemoryPolynomialConfiguration
+                {
+                    MemoryModelCorrectionType = RFmxSpecAnMXDpdApplyDpdMemoryModelCorrectionType.MagnitudeAndPhase,
+                    IterativeDpdEnabled = RFmxSpecAnMXDpdIterativeDpdEnabled.False,
+                    NumberOfIteration = 3,
+                    MemoryPolynomialDepth = 2,
+                    MemoryPolynomialOrder = 3,
+                    MemoryPolynomialLeadOrder = 2,
+                    MemoryPolynomialLagOrder = 2,
+                    MemoryPolynomialLeadMemoryDepth = 2,
+                    MemoryPolynomialLagMemoryDepth = 2,
+                    MemoryPolynomialMaxLead = 2,
+                    MemoryPolynomialMaxLag = 2
+                };
+            }
+        }
+
+        public struct LookupTableResults
+        {
+            public ComplexWaveform<ComplexSingle> WaveformWithDpdComplexSingle;
```

Comment:
I'm wondering if it might be wise to return the SG.Waveform struct instead of just the waveform itself in case it is needed for later automation purposes.  We fill out the whole struct in the Apply functions already since it is passed to the SG module. What do you think @smooresni ?
```suggestion
            public SG.Waveform PostDPDWaveform;
```

Reply:
@mdwhitten agreed